server.js

```diff
@@ -1,33 +1,58 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// Defines the port the app will run on. Defaults to 8080
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// First page
 app.get('/', (req, res) => {
-  res.send('Hello world')
+	res.send('This is an API with books')
+})
+// All books in the API
+app.get('/books', (req, res) => {
+	const { title, author, sort } = req.query
+	let bookList = booksData
+
+	// Filtering by title
+	if (title) {
+		bookList = bookList.filter((item) => item.title.toString().toLowerCase().includes(title.toLocaleLowerCase()))
+	}
+	// Filtering by Author
+	if (author) {
+		bookList = bookList.filter((item) => item.authors.toString().toLowerCase().includes(author.toLocaleLowerCase()))
+	}
+	// Rating high to low
+	if (sort === 'rating_dsc') {
+		bookList = bookList.sort((a, b) => b.average_rating - a.average_rating)
+		//Rating low to high
+	} else if (sort === 'rating_asc') {
+		bookList = bookList.sort((a, b) => a.average_rating - b.average_rating)
+	}
+	// Error message for title & author
+	if (bookList.length === 0) res.status(404).send({ error: 'Sorry, could not find that title/author' })
+	res.json({ numberOfBooks: bookList.length, result: bookList })
 })
+// Endpoint ID
+app.get('/books/:id', (req, res) => {
+	const id = req.params.id
+	const showBook = booksData.find((item) => item.bookID === +id)
 
-// Start the server
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`)
+	// Error message for ID
+	if (!showBook) res.status(404).send({ error: 'Sorry, could not find that book' })
```

Comment:
Not very readable - I agree that there is a use case for if statement without curly brackets, but I would argue if it's this case